numba/tests/test_typedlist.py

```diff
@@ -106,6 +106,17 @@ def test_basic(self):
         # index
         self.assertEqual(l.index(15), 4)
 
+    def test_list_extend_unicode_type(self):
```

Comment:
Maybe you could call this:

```suggestion
    def test_list_extend_refines_on_unicode_type(self):
```

to emphasize that this test checks refinement?